multicluster/controllers/multicluster/commonarea/remote_common_area_test.go

```diff
@@ -79,9 +81,189 @@ func TestMemberAnnounce(t *testing.T) {
+
+	assert.Equal(t, v1.ConditionTrue, status[0].Status)
+	assert.Equal(t, v1.ConditionTrue, status[1].Status)
+	assert.Equal(t, "", status[0].Reason)
+	assert.Equal(t, "", status[1].Reason)
+	assert.Equal(t, "", status[0].Message)
+	assert.Equal(t, "This leader cluster is the leader for local cluster", status[1].Message)
+	assert.Equal(t, mcsv1alpha1.ClusterConditionType(""), status[0].Type)
+	assert.Equal(t, mcsv1alpha1.ClusterIsLeader, status[1].Type)
+}
+
+func TestMemberAnnounceNewRemoteCommonArea(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	mockManager := mocks.NewMockManager(mockCtrl)
+	fakeRemoteClient := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+	expectedRemoteCommonArea := &remoteCommonArea{
+		Client:             fakeRemoteClient,
+		ClusterManager:     mockManager,
+		ClusterSetID:       "clusterSetA",
+		ClusterID:          "leaderA",
+		config:             nil,
+		scheme:             scheme,
+		Namespace:          "cluster-a-ns",
+		connected:          false,
+		localClusterClient: nil,
+		localNamespace:     "localnamespace",
+		localClusterID:     "clusterA",
+		clusterStatus: mcsv1alpha1.ClusterCondition{
+			Type:    mcsv1alpha1.ClusterReady,
+			Status:  v1.ConditionUnknown,
+			Message: "Leader cluster added",
+		},
+		leaderStatus: mcsv1alpha1.ClusterCondition{
+			Type:    mcsv1alpha1.ClusterIsLeader,
+			Status:  v1.ConditionFalse,
+			Message: "Leader cluster added",
+		},
+	}
+
+	actualRemoteCommonArea, err := NewRemoteCommonArea(expectedRemoteCommonArea.ClusterID, expectedRemoteCommonArea.ClusterSetID, expectedRemoteCommonArea.localClusterID, mockManager, fakeRemoteClient, scheme, nil,
+		"cluster-a-ns", "localnamespace", nil)
+	assert.Equal(t, nil, err)
+	clusterStatus, leaderStatus := actualRemoteCommonArea.GetStatus()[0], actualRemoteCommonArea.GetStatus()[1]
+	expectedRemoteCommonArea.clusterStatus.LastTransitionTime = clusterStatus.LastTransitionTime
+	expectedRemoteCommonArea.leaderStatus.LastTransitionTime = leaderStatus.LastTransitionTime
+	assert.Equal(t, expectedRemoteCommonArea, actualRemoteCommonArea)
+}
+
+func TestMemberAnnounceGetSecretCACrtAndToken(t *testing.T) {
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "mcs1",
+			Name:      "member-token",
+		},
+		Data: map[string][]byte{
+			"ca.crt": []byte(`12345`),
+			"token":  []byte(`12345`)},
+	}
+	secretNoRootCAKey := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "mcs1",
+			Name:      "member-token",
+		},
+		Data: map[string][]byte{
+			"token": []byte(`12345`)},
+	}
+	secretNoTokenKey := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "mcs1",
+			Name:      "member-token",
+		},
+		Data: map[string][]byte{
+			"ca.crt": []byte(`12345`),
+		},
+	}
+
+	tests := []struct {
+		name          string
+		secret        *v1.Secret
+		caData        []byte
+		token         []byte
+		expectedError error
+	}{
+		{
+			name:   "with no conditions",
```

Comment:
Please update these test case's names, seems you copied from other tests but forgot to update the name.